### PR TITLE
Version 15.0.0 → 16.0.0

### PR DESCRIPTION
**Pull request by Gabriel439**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,106 @@ file.
...
+  Now, a union type like this one is valid:
+
+  ```dhall
+  < x : Bool | y | z : Type > 
+  ```
+
+  ... or more generally you can mix terms, types, and kinds within the same
+  union (similar to how you can mix them within records).
+
+  Besides permitting a more expressions this change also simplifies the
```
Comment (travisbrown) [inline CHANGELOG.md:89]: Minor typo: "a more".

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,106 @@ file.
...
+    This part is not a breaking change because any expression affected by this
+    change in precedence would not have type-checked anyway.
+
+* [Update encoding of floating point values to RFC7049bis](https://github.com/dhall-lang/dhall-lang/pull/958)
+
+  This changes how `Double` literals are encoded to match a new standard
+  recommendation how to canonically encode floating point values.  Specifically,
+  these values are now encoded using the smallest available CBOR representation,
+  (which could be half precision floating point values).  Previously, the Dhall
+  standard will always use at least 32 bits for encoding floating point values.
```
Comment (travisbrown) [inline CHANGELOG.md:67]: Probably doesn't matter, but this isn't entirely accurate, since the zeros and infinities and NaNs always only used 16 bits.

Comment (philandstuff) [reply CHANGELOG.md:67]:
```suggestion
  standard would always use at least 32 bits for encoding floating point values.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,106 @@ file.
...
+    Previously, if you omitted the parentheses, like this:
+
+    ```dhall
+    { x = 0 } // { y = 1 } with x = 1
+    ```
+
+    ... that would have parsed as:
+
+    ```dhall
+    { x = 0 } // ({ y = 1 } with x = 1)
```
[the remark this answers is not in the document]
Comment (Nadrieril) [reply CHANGELOG.md:33]: It could make a difference if we used `/\`

**Comment by Nadrieril** [comment]
@Gabriel439 There's another breaking change related to `with`:
```dhall
foo with x.y = {} // {}
```
The `//` used to be parsed as outside the `with`, but now is parsed as inside of it.
This is so that `foo with x.y = 1+1` parses as expected.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,135 @@ file.
...
+
+    This restriction forbids expressions that might have been ambiguous to
+    readers.  This is definitely a breaking change, since such expressions were
+    previously valid now require explicit parentheses, otherwise they will fail
+    to parse.
+
+    Along the same lines, an expression like this:
+
+    ```dhall
+    record with x.y = {} // {}
```
Comment (philandstuff) [inline CHANGELOG.md:47]:
```suggestion
    record with x.y = {} // { x = 1 }
```

**Comment by Gabriel439** [comment]
Note that I will likely also include:

* [ ] https://github.com/dhall-lang/dhall-lang/pull/971
* [ ] https://github.com/dhall-lang/dhall-lang/pull/972
* [ ] https://github.com/dhall-lang/dhall-lang/pull/983

... since they are like to merge within the 3-day waiting period for this

**Pull request merged**